### PR TITLE
Remove weak ciphers from dovecot 10-ssl.conf

### PR DESCRIPTION
**Pull request by neto737**
[TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA](https://ciphersuite.info/cs/TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA/)
[ECDHE-RSA-DES-CBC3-SHA](https://ciphersuite.info/cs/TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA/)
[AES128-GCM-SHA256](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_128_GCM_SHA256/)
[AES256-GCM-SHA384](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_256_GCM_SHA384/)
[AES128-SHA256](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_128_CBC_SHA256/)
[AES256-SHA256](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_256_CBC_SHA256/)
[AES128-SHA](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_128_CBC_SHA/)
[AES256-SHA](https://ciphersuite.info/cs/TLS_RSA_WITH_AES_256_CBC_SHA/)

### CODE REVIEW

File: install/common/dovecot/conf.d/10-ssl.conf [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+ssl_cipher_list = ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:EDH-RSA-DES-CBC3-SHA:!DSS
```
Comment (myrevery) [inline install/common/dovecot/conf.d/10-ssl.conf]: Please remove wrong cipher: "EDH-RSA-DES-CBC3-SHA"

**Comment by myrevery** [comment]
Also, we should adjust the cipher list order.

And add "DHE-RSA-CHACHA20-POLY1305" to the list.

Suggested options:
1. Remove all SHA1 for enhanced security (full FS).
```
ssl_cipher_list = ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA256:!DSS

ECDHE-ECDSA-CHACHA20-POLY1305
ECDHE-RSA-CHACHA20-POLY1305
DHE-RSA-CHACHA20-POLY1305
ECDHE-ECDSA-AES128-GCM-SHA256
ECDHE-RSA-AES128-GCM-SHA256
DHE-RSA-AES128-GCM-SHA256
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
DHE-RSA-AES256-GCM-SHA384
ECDHE-ECDSA-AES128-SHA256
ECDHE-RSA-AES128-SHA256
DHE-RSA-AES128-SHA256
ECDHE-ECDSA-AES256-SHA384
ECDHE-RSA-AES256-SHA384
DHE-RSA-AES256-SHA256
```
2. With SHA1
```
ssl_cipher_list = ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA256:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA:!DSS

ECDHE-ECDSA-CHACHA20-POLY1305
ECDHE-RSA-CHACHA20-POLY1305
DHE-RSA-CHACHA20-POLY1305
ECDHE-ECDSA-AES128-GCM-SHA256
ECDHE-RSA-AES128-GCM-SHA256
DHE-RSA-AES128-GCM-SHA256
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
DHE-RSA-AES256-GCM-SHA384
ECDHE-ECDSA-AES128-SHA256
ECDHE-RSA-AES128-SHA256
DHE-RSA-AES128-SHA256
ECDHE-ECDSA-AES128-SHA
ECDHE-RSA-AES128-SHA
DHE-RSA-AES128-SHA
ECDHE-ECDSA-AES256-SHA384
ECDHE-RSA-AES256-SHA384
DHE-RSA-AES256-SHA256
ECDHE-ECDSA-AES256-SHA
ECDHE-RSA-AES256-SHA
DHE-RSA-AES256-SHA
```

**Comment by neto737** [comment]
As we're going to disable TLSv1.0 and TLSv1.1 I don't think we need to keep ciphers that are supported only in these versions of TLS.

Using [CipherSuite](https://ciphersuite.info/) API  I was able to get a list of all ciphers that are supported by TLSv1.2 and TLSv1.3 and also are still considered "secure" or "recommended" by them.

Here's the list, if you agree with it, I'll edit the 10-ssl.conf with these ciphers instead of the old ones.

```
TLS_AES_128_CCM_SHA256
TLS_AES_128_GCM_SHA256
TLS_AES_256_GCM_SHA384
DHE-DSS-AES128-GCM-SHA256
DHE-DSS-AES256-GCM-SHA384
DHE-PSK-AES128-CCM
DHE-PSK-AES128-GCM-SHA256
DHE-PSK-AES256-CCM
DHE-PSK-AES256-GCM-SHA384
DHE-PSK-CHACHA20-POLY1305
DHE-RSA-AES128-CCM
DHE-RSA-AES128-CCM8
DHE-RSA-AES128-GCM-SHA256
DHE-RSA-AES256-CCM
DHE-RSA-AES256-CCM8
DHE-RSA-AES256-GCM-SHA384
DHE-RSA-CHACHA20-POLY1305
ECDHE-ECDSA-AES128-CCM
ECDHE-ECDSA-AES128-CCM8
ECDHE-ECDSA-AES128-GCM-SHA256
ECDHE-ECDSA-AES256-CCM
ECDHE-ECDSA-AES256-CCM8
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-ECDSA-CHACHA20-POLY1305
ECDHE-PSK-CHACHA20-POLY1305
ECDHE-RSA-AES128-GCM-SHA256
ECDHE-RSA-AES256-GCM-SHA384
ECDHE-RSA-CHACHA20-POLY1305
DHE-PSK-AES128-CCM8
DHE-PSK-AES256-CCM8
```

**Comment by neto737** [comment]
After updating nginx.conf to use only secure ciphers I decided to test it on [SSLLabs](https://www.ssllabs.com/ssltest/) and noticed that some (old) browsers don't have support to secure ciphers.

The "most secure" if we can call it this way, for these browsers, is _TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (ECDHE-ECDSA-AES256-SHA384)_

- [Safari 8 / OS X 10.10](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=8&platform=OS%20X%2010.10&key=87)
- [Safari 8 / iOS 8.4](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=8&platform=iOS%208.4&key=85)
- [Safari 7 / OS X 10.9](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=7&platform=OS%20X%2010.9&key=35)
- [Safari 7 / iOS 7.1](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=7&platform=iOS%207.1&key=63)
- [Safari 6 / iOS 6.0.1](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=6&platform=iOS%206.0.1&key=33)
- [IE 11 / Win Phone 8.1](https://www.ssllabs.com/ssltest/viewClient.html?name=IE&version=11&platform=Win%20Phone%208.1&key=65)

Should we keep this weak cipher to keep support to these outdated browsers?

**Comment by myrevery** [comment]
IMHO, please stay focused on Dovecot, and be more like an expert.

But if you're not...

First, the ciphers order can't be messed set.

Requirements:
Secure priority, performance priority and COMPATIBILITY.

I won't repeat the literal meaning of "ssl_prefer_server_ciphers = yes", please forgive me...

And Exim doesn't support setting TLSv1.3 ciphers, see: [Exim doc](https://www.exim.org/exim-html-current/doc/html/spec_html/ch-encrypted_smtp_connections_using_tlsssl.html#:~:text=For%20TLS%20version%201.3%20the%20control%20available%20is%20less%20fine%2Dgrained%20and%20Exim%20does%20not%20provide%20access%20to%20it%20at%20present.%20The%20value%20of%20the%20tls_require_ciphers%20option%20is%20ignored%20when%20TLS%20version%201.3%20is%20negotiated.)

The "weak" ciphers are NOT "broken" ciphers, making the ciphers list full FS is now seucre enough.

You mentioned SSL Labs, so I suggest you check Cloudflare or Google's ciphers.

Please remember, most servers managed by Hestia should be for providing public services, COMPATIBILITY is very important!

**Comment by neto737** [comment]
> IMHO, please stay focused on Dovecot, and be more like an expert.
> 
> But if you're not...
> 
> First, the ciphers order can't be messed set.
> 
> Requirements: Secure priority, performance priority and COMPATIBILITY.
> 
> I won't repeat the literal meaning of "ssl_prefer_server_ciphers = yes", please forgive me...
> 
> And Exim doesn't support setting TLSv1.3 ciphers, see: [Exim doc](https://www.exim.org/exim-html-current/doc/html/spec_html/ch-encrypted_smtp_connections_using_tlsssl.html#:~:text=For%20TLS%20version%201.3%20the%20control%20available%20is%20less%20fine%2Dgrained%20and%20Exim%20does%20not%20provide%20access%20to%20it%20at%20present.%20The%20value%20of%20the%20tls_require_ciphers%20option%20is%20ignored%20when%20TLS%20version%201.3%20is%20negotiated.)
> 
> The "weak" ciphers are NOT "broken" ciphers, making the ciphers list full FS is now seucre enough.
> 
> You mentioned SSL Labs, so I suggest you check Cloudflare or Google's ciphers.
> 
> Please remember, most servers managed by Hestia should be for providing public services, COMPATIBILITY is very important!

Agreed, let's focus on Dovecot

**Comment by myrevery** [comment]
Sorry I just woke up, I'll give more info later...

**Comment by myrevery** [comment]
@neto737 @jaapmarcus Sorry, I'm too busy...

I maintain a private fork of Hestia to serve my customers, and the built-in "Enhanced and Optimized TLS" is one of the paid features.

Now I decided to share it upstream.

ETA within this week, If you guys don't mind, this PR needs to be closed upon arrival.

**Comment by jaapmarcus** [comment]
Replaced by:
https://github.com/hestiacp/hestiacp/pull/3555

**Pull request closed without merging**